x-pack/plugin/relevance-search/src/test/java/org/elasticsearch/xpack/relevancesearch/query/RelevanceMatchQueryIntTests.java

```diff
@@ -26,50 +27,99 @@
-            )
-            .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
-            .execute()
-            .actionGet();
+    @Override
+    public void setUp() throws Exception {
+        super.setUp();
+        relevanceMatchQueryBuilder = new RelevanceMatchQueryBuilder();
+        relevanceMatchQueryBuilder.setRelevanceSettingsService(getInstanceFromNode(RelevanceSettingsService.class));
+
+        createIndex(DOCUMENTS_INDEX);
+        createIndex(RelevanceSettingsService.ENT_SEARCH_INDEX);
+    }
+
+    public void testTextFieldsWitoutSettings() {
+        indexDocument(
+            DOCUMENTS_INDEX,
+            "1",
+            Map.of("textField", "text example", "intField", 12, "doubleField", 13.45, "anotherTextField", "should match")
+        );
+        indexDocument(
+            DOCUMENTS_INDEX,
+            "2",
+            Map.of("textField", "other document", "intField", 12, "doubleField", 13.45, "anotherTextField", "should not be found")
+        );
 
-        RelevanceMatchQueryBuilder relevanceMatchQueryBuilder = new RelevanceMatchQueryBuilder();
         relevanceMatchQueryBuilder.setQuery("text match");
-        SearchResponse response = client().prepareSearch("index").setQuery(relevanceMatchQueryBuilder).get();
+        SearchResponse response = client().prepareSearch(DOCUMENTS_INDEX).setQuery(relevanceMatchQueryBuilder).get();
 
         assertHitCount(response, 1);
         assertSearchHits(response, "1");
     }
 
-    public void testRelevanceMatchQueryWithNoTextFields() {
-        client().admin().indices().prepareCreate("index").get();
-        client().prepareIndex("index")
-            .setId("1")
-            .setSource(Map.of("intField", 12, "doubleField", 13.45))
-            .setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE)
-            .execute()
-            .actionGet();
+    public void testNoTextFieldsWithoutSettings() {
+
+        indexDocument(DOCUMENTS_INDEX, "2", Map.of("intField", 12, "doubleField", 13.45));
 
-        RelevanceMatchQueryBuilder relevanceMatchQueryBuilder = new RelevanceMatchQueryBuilder();
         relevanceMatchQueryBuilder.setQuery("text match");
-        final SearchRequestBuilder searchRequestBuilder = client().prepareSearch("index").setQuery(relevanceMatchQueryBuilder);
+        final SearchRequestBuilder searchRequestBuilder = client().prepareSearch(DOCUMENTS_INDEX).setQuery(relevanceMatchQueryBuilder);
 
         final String expectedMsg = "[relevance_match] query cannot find text fields in the index";
         assertFailures(searchRequestBuilder, RestStatus.BAD_REQUEST, containsString(expectedMsg));
     }
+
+    public void testFieldSettings() {
+        final String settingsId = "test-settings";
+        indexDocument(
+            RelevanceSettingsService.ENT_SEARCH_INDEX,
+            RelevanceSettingsService.RELEVANCE_SETTINGS_PREFIX + settingsId,
+            Map.of("fields", List.of("textField"))
```

Comment:
I think it would be good to add field weights here, for completeness. WDYT?

Reply:
Good call, thanks! Added.